node/network/bitfield-distribution/src/lib.rs

```diff
@@ -674,22 +677,28 @@ mod test {
 		}
 	}
 
-	fn state_with_view(view: View, relay_parent: Hash) -> (ProtocolState, SigningContext, ValidatorPair) {
+	fn state_with_view(
+		view: View,
+		relay_parent: Hash,
+		keystore_path: &tempfile::TempDir,
+	) -> (ProtocolState, SigningContext, CryptoStorePtr, ValidatorId) {
 		let mut state = ProtocolState::default();
 
-		let (validator_pair, _seed) = ValidatorPair::generate();
-		let validator = validator_pair.public();
-
 		let signing_context = SigningContext {
 			session_index: 1,
 			parent_hash: relay_parent.clone(),
 		};
 
+		let keystore : CryptoStorePtr = Arc::new(LocalKeystore::open(keystore_path.path(), None)
+			.expect("Creates keystore"));
+		let validator = executor::block_on(keystore.sr25519_generate_new(ValidatorId::ID, None))
+			.expect("Malicious key created");
```

Comment:
```suggestion
			.expect("generating sr25519 key not to fail");
```

What makes this key malicious?